tests/tests/test-xcm/test-xcmv3-max-weight-instructions.ts

```diff
@@ -0,0 +1,222 @@
+import "@moonbeam-network/api-augment";
+import { expect } from "chai";
+import { CHARLETH_ADDRESS } from "../../util/accounts";
+import { describeDevMoonbeam } from "../../util/setup-dev-tests";
+import {
+  XcmFragment,
+  injectHrmpMessage,
+  RawXcmMessage,
+  sovereignAccountOfSibling,
+} from "../../util/xcm";
+import { expectEVMResult } from "../../util/eth-transactions";
+import { ALITH_TRANSACTION_TEMPLATE, createTransaction } from "../../util/transactions";
+
+describeDevMoonbeam(
+  "XCM V3 - Max Weight Instructions",
+  (context) => {
+    let dotAsset: any;
+    let amount: bigint;
+    let paraId: number;
+
+    before("Set up initial constants", async function () {
+      paraId = 888;
+      const paraSovereign = sovereignAccountOfSibling(context, paraId);
+      const metadata = await context.polkadotApi.rpc.state.getMetadata();
+      const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
```

Comment:
you don't need to `toHuman()` this as type `PalletMetadataV15` should be correctly inferred. e.g.
```
(await context.polkadotApi.rpc.state.getMetadata()).asLatest.pallets.find(({name})=> name.toString() === "balances")!.index.toNumber()
```